app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -167,7 +167,12 @@ class TransactionStateManager extends EventEmitter {
         transactions.splice(index, 1)
       }
     }
-    transactions.push(txMeta)
+    const newTxIndex = transactions
+      .findIndex((currentTxMeta) => currentTxMeta.id > txMeta.id)
+
+    newTxIndex === -1
+      ? transactions.push(txMeta)
+      : transactions.splice(newTxIndex, 0, txMeta)
```

Comment:
if this is a new tx why the splice?

Reply:
this inserts the transaction in the list according to its `id`, before the first transaction with a greater  `id`